### ISSUE TITLE
Page scripts cdn support issues

### ISSUE THREAD

**Issue by lyt9304** [zeit/next.js#3916]
### Environment

- mac osx 10.13.3
- next.js 4.1.4
- react 16.2.0

#### What I want to do

I want to upload my resource files to cdn for performance concern. 

resource files such as:

- 1.static files like

	- `/static/images/logo.png`
	- `/static/js/slides.min.js`

- 2.compiled scripts
	- page scripts like `/_next/d40f48be-1234-499c-bb8a-652b581c8f55/page/`
	- other scripts like `/_next/d40f48be-c687-499c-bb8a-652b581c8f55/page/_error/index.js` 

### Problem I met

First one is easy to handle, but second one seems complicated.

With inspires of [#3621](https://github.com/zeit/next.js/issues/3621), I found that `next export` will handle this transform for us, so we don't have to worry about how actual dist files map to `/_next/${hash}/page/${filename}` structure.

So I tried to use `next export` to generate files.

```js
// next.config.js

module.exports = {
	assetPrefix: 'https://some-cdn.net',
	exportPathMap: () => {
		return {}
	},
}
```

And then run the export command

```bash
npx next export ./dist/client -o ./export
```

So In the export directory, next generate the file structure exactly match the pattern like `/_next/${hash}/page/${filename}/index.js`, and that's what I wanted. So I can just upload the `_next` directory to cdn.

![2018-02-28 4 53 42](https://user-images.githubusercontent.com/5561692/36778828-e76804de-1ca8-11e8-93b7-23128528d6fa.jpg)

So far it seems good, but I actually run the site, and I got this result.
![1519808138935](https://user-images.githubusercontent.com/5561692/36778849-faef145c-1ca8-11e8-9c52-379c5a3d4d66.jpg)
![1519808168943](https://user-images.githubusercontent.com/5561692/36778852-fdfd5be0-1ca8-11e8-8281-7887b2f3d9f2.jpg)

Because when next generate the html document, it will write `/page` instead of `/page/index.js`.

If I'm not using cdn to serve my scripts and let next serve this file, next server will know that it should return `index.js`, but my cdn file don't know how to handle this request.

I just read the code of `server/document.js`, there is a function seems to do the job like append index.js to the path.

```js
function getPagePathname (pathname, nextExport) {
  if (!nextExport) return pathname
  if (pathname === '/') return '/index.js'
  return `${pathname}/index.js`
}
```

But the question is my web site is not a static site, it will request many api to get the data of page before render it (getInitialProps), so I can't just export it.

Is there any setting or method that I can add `/index.js` to page scripts src, and not just request like `/page`?

I'm curious about why not all just write `/page/index.js` to avoid ambiguity.

Thank you for your time and help! : )

### Other

I think it's not very friendly for cdn support, and document only mentioned `assetPrefix`, it will take a lot of work to get the result, maybe document can apply some best practice of cdn support, so we can just follow it?

**Comment by timneutkens** [comment]
Try upgrading to next 5, the second issue is fixed there. The first is fixed in next@canary (upcoming 5.1 release).

**Comment by vaibhavi3t** [comment]
@lyt9304 

I have the same issue. Have you got the solution? Please explain here.
